### PR TITLE
Combine char restrictions

### PR DESCRIPTION
**Pull request by Da-Colon**
## Description
replaces the char restriction props into a single ENUM controlled prop

## Notes
Solves issue where the Input component can be implemented with `isWholeNumbers` and `isFloatNumbers`, This will cause the two methods to conflict with each other.

## Issue (if applicable)

discussed in #474

## Testing

no Visual QA is needed. More just a technical review

## Screenshots (if applicable)
n/a

### CODE REVIEW

File: src/components/ui/forms/Input.tsx [hunk 1]
```diff
@@ -139,17 +142,18 @@ function Input({
...
-      }
       if (decimals) {
         limitDecimals(event);
       }
+      if (restrictChar === RestrictCharTypes.WHOLE_NUMBERS_ONLY) {
+        wholeNumbersOnly(event);
+      }
+      if (restrictChar === RestrictCharTypes.FLOAT_NUMBERS) {
+        floatNumbersOnly(event);
+      }
```
Comment (adamgall) [inline src/components/ui/forms/Input.tsx:153]: nitpick

The interface suggests that only one of these types can be set on an instance of `Input`, but this implementation is written in a way that, at first glance, suggests that both of those `if` blocks can be entered.

Can the implementation here be changed to a `switch`, which clearly shows that only ONE of the types will be enforced?

**Review by adamgall — APPROVED** [review]
Approving this, but also asking for a small change which is very nitpicky

File: src/components/ui/forms/Input.tsx (hunk 1 above)
Comment (Da-Colon) [reply src/components/ui/forms/Input.tsx:153]: Ha i had thought about it but didnt. For sure i will get added before merging

Comment (adamgall) [reply src/components/ui/forms/Input.tsx]: I did this in commit https://github.com/decent-dao/fractal-interface/pull/482/commits/f7e8cc954e0bec787d0147aacc66eb5f50ed5947

**Pull request merged**